### ISSUE TITLE
Passing a subdomain via custom_include.vcl does not seem to work

### ISSUE THREAD

**Issue by mobweb** [nexcess/magento-turpentine#1083]
I have added the following to _custom_include.vcl_:

sub vcl_recv {
	if (req.http.host ~ "staging") {
	   return (pass);
	}
}

This does not seem to work, as I am still getting hits when accessing the "staging" subdomain.

However, if I manually overwrite the Varnish configuration as generated by Turpentine and add my "_if_"-condition at the beginning of the existing "_vcl_recv_" subroutine, I am no longer getting any hits on the "staging" subdomain.

My question is this: Is it possible that my custom "_if_" condition is never executed because _custom_include.vcl_ is added at the very bottom of the VCL file as created by Turpentine?

**Comment by aricwatson** [comment]
Do you mind adding the include to the top to see if it works in that scenario?

**Comment by mobweb** [comment]
Yes, sorry.

If I move the _{{custom_vcl_include}}_ to right before where _sub vcl_recv_ is first mentioned in _version-3.vcl_ and then apply the configuration it is working as intended, and the subdomain is excluded from the cache.

I also noticed that in the Wiki it says this:

> Custom VCL File Location: Set the location for the custom VCL include file. The contents of this file will be included **at the top** of the generated VCL if it exists

**Comment by aricwatson** [comment]
Yes, we recently changed it to the bottom - since it worked better for a different issue to be there. Sounds like we'll need to make it configurable whether the custom_include goes at top or bottom. Thanks for the wiki find, I'll fix that.

**Comment by ADDISON74** [comment]
@aricwatson: in this scenario you have to think about having two or more custom files to avoid any issues. it could be one at the top, or bottom, or giving the ability to edit these files content in backend which is a more elegant way. for example choosing your varnish version file then editing in backend. these changes will be effective when creating/changing Varnish configuration file and will skip editing files in a terminal.

**Comment by mobweb** [comment]
Yes, I also thought that for my issue, having one file at the top and one at the bottom would be best.

However, I disagree that the file's contents should be manageable from the backend, as I personally prefer being able to use version tracking for as many settings as possible (which is obviously not possible if the configuration is stored in the DB).

**Comment by aricwatson** [comment]
I will add configuration options to allow custom include files at top *and/or* bottom of the generated file. I agree with @mobweb that it's best not to add editing capabilities for these files into the Turpentine back-end.
